File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/objects/schedule/ScheduleItem.java

package de.markhaehnel.rbtv.rocketbeanstv.objects.schedule;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import de.markhaehnel.rbtv.rocketbeanstv.utils.Time;
import lombok.Getter;

@Getter
public class ScheduleItem {

    @SerializedName("id")
    @Expose
    private Long id;
    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("topic")
    @Expose
    private String topic;
    @SerializedName("show")
    @Expose
    private String show;
    @SerializedName("timeStart")
    @Expose
    private String timeStart;
    @SerializedName("timeEnd")
    @Expose
    private String timeEnd;
    @SerializedName("length")
    @Expose
    private Long length;
    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("game")
    @Expose
    private String game;

    public String getTimeStartShort() {
        return Time.getShortTimeFromISO(getTimeStart());
    }
}

File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/objects/schedule/ScheduleItem.kt
package de.markhaehnel.rbtv.rocketbeanstv.objects.schedule

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import de.markhaehnel.rbtv.rocketbeanstv.utils.Time

class ScheduleItem {

    @SerializedName("id")
    @Expose
    val id: Long? = null

    @SerializedName("title")
    @Expose
    val title: String? = null

    @SerializedName("topic")
    @Expose
    val topic: String? = null

    @SerializedName("show")
    @Expose
    val show: String? = null

    @SerializedName("timeStart")
    @Expose
    val timeStart: String? = null

    @SerializedName("timeEnd")
    @Expose
    val timeEnd: String? = null

    @SerializedName("length")
    @Expose
    val length: Long? = null

    @SerializedName("type")
    @Expose
    val type: String? = null

    @SerializedName("game")
    @Expose
    val game: String? = null

    val timeStartShort: String
        get() = Time.getShortTimeFromISO(timeStart!!)
}
